Fix race when receiving docker die event

tgraf opened the pull request:
The IPAM unregistration is currently bound to the docker die event.
This leaves IPAM addresses registered if the endpoint deletion
occurs through the API instead of the docker event.

aanm commented on daemon/daemon/endpoint.go:
```diff
@@ -26,6 +26,8 @@ import (
 	"github.com/cilium/cilium/pkg/labels"
 	"github.com/cilium/cilium/pkg/option"
 	"github.com/cilium/cilium/pkg/policy"
+
```
remove new line

aanm reviewed (changes requested):
Just the remove new line

tgraf committed:
daemon: Fix race when receiving docker die event

The IPAM unregistration is currently bound to the docker die event.
This leaves IPAM addresses registered if the endpoint deletion
occurs through the API instead of the docker event.

Signed-off-by: Thomas Graf <thomas@cilium.io>
